flake.nix

```diff
@@ -24,20 +24,57 @@
       rec {
         packages = rec {
           default = merlin;
-          merlin = buildDunePackage {
-            pname = "merlin";
-            version = "n/a";
+          merlin-lib = buildDunePackage {
+            pname = "merlin-lib";
+            version = "dev";
             src = ./.;
             duneVersion = "3";
-            buildInputs = with pkgs.ocamlPackages; [ menhirSdk menhir ];
             propagatedBuildInputs = with pkgs.ocamlPackages; [
-              findlib
               csexp
-              yojson
+            ];
+            doCheck = true;
+          };
+          dot-merlin-reader = buildDunePackage {
+            pname = "dot-merlin-reader";
+            version = "dev";
+            src = ./.;
+            duneVersion = "3";
+            buildInputs = with pkgs.ocamlPackages; [
+              merlin-lib
```

Comment:
this is probably using `merlin-lib` from `ocamlPackages`, if it exists, rather than the recursive one here.

Reply:
fixed